src/gmpints.c

```diff
@@ -375,6 +375,31 @@ Obj ObjInt_UInt( UInt i )
   }
 }
 
+Obj ObjInt_Int8( Int8 i )
+{
+  int neg = 0;
+  if(i < 0) {
+    neg = 1;
+    i = -i;
+  }
```

Comment:
Actually, it might not be as bad as I thought, because the additive inverse of -2^63 is itself (for Int8 values, that is). I still think avoiding the overhead is a good idea.